docs/source/single_page_run_through.rst

```diff
@@ -0,0 +1,400 @@
+The result is just a normal python function, so you call it with its expected arguments::
+
+    result = rolling_ols(x, 10)
+    large_window_result = rolling_ols(x, 250)
+
+The `@fs` decorator
+^^^^^^^^^^^^^^^^^^^
+
+The ability to pass functions as arguments to numba functions and have everything jitted correctly comes from the @fs
+decorator. When this decorator is applied to a function, it changes the semantics of keyword arguments;
+keyword arguments indicate that any function calls of that argument name should be replaced with the function
+being passed in. For example::
+
+    def mean(x):
+        return np.sum(x) / x.size
+
+    result = single_pass(x, value=mean)
+
+The definition of `single_pass` does not take any keyword arguments, or specify `**kwargs`, but it has an `@fs`
+decorator and therefore substitutes all occurences of `value` for `mean` in its own function body, and in the function
+bodies of any of its child functions.
+
+.. automodule:: fastats.core.decorator
+    :members:
+
+fastats.linear_algebra
+----------------------
+
+.. automodule:: fastats.linear_algebra
+    :members:
+
+fastats.maths
+-------------
+
+.. automodule:: fastats.maths
+    :members:
+
+fastats.optimise
+----------------
+
+.. automodule:: fastats.optimise
+    :members:
+
+
+Technical Details
+-----------------
+
+How do the AST Transforms work?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Imagine a nested function such as a very simple Newton-Raphson solver::
+
+    def newton_raphson(x0, delta, root=fs_func, deriv=deriv):
+        last_x = x0
+        next_x = last_x + 10 * delta
+        while abs(last_x - next_x) > delta:
+            new_y = root(next_x)
+            last_x = next_x
+            next_x = last_x - new_y / deriv(last_x, delta)
+        return next_x
+
+
+In the code above we actually want to replace the `root` function with one of our own
+choosing, without having to re-write the entire newton_raphson wrapper function.
+
+In `fastats` this is performed by changing the semantics of positional and keyword
+arguments; `numba` does not allow us to pass functions as arguments, but even if it
+did, we still need the ability to arbitrarily modify deeply nested functions (use
+cases discussed below).
+
+The Newton-Raphson code shown above is in the `fastats` standard library, and allows us
+to do the following::
+
+    from fastats.optimise.root_finding.newton_raphson import newton_raphson
+
+    def my_func(x, y):
+        return x**2 + y**2
+
+    my_solver = newton_raphson(, 0.001, root=my_func)
+
+    my_solver(3, 4)
+
+`my_func` is the function for which we would like to find the roots. It takes two arguments,
+`x` and `y`, one of which will be kept constant whilst the other is varied to find the root.
+
+When `newton_raphson` is called, it takes the `root=my_func` kwarg, and inspects the signature
+of `my_func`. It finds that my_func takes 2 arguments `x` and `y`, and expects the first argument
```

Comment:
Trivial, but should my_func be code formatted here too?

Reply:
👍